source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -879,6 +879,11 @@ void Filter::onComplete(Filters::Common::ExtAuthz::ResponsePtr&& response) {
       return;
     }
 
+    if (request_headers_->byteSize() > request_headers_->maxHeadersKb() * 1024) {
```

Comment:
nit: can you merge this if check into the check above `if (request_headers_->size() > request_headers_->maxHeadersCount()) `